hazelcast/src/main/java/com/hazelcast/map/impl/MapContainer.java

```diff
@@ -284,6 +286,10 @@ public boolean isInvalidationEnabled() {
     public InterceptorRegistry getInterceptorRegistry() {
         return interceptorRegistry;
     }
+
+    public boolean canBeRemoved() {
```

Comment:
a very small thing: something like `markForRemoval()` could be better. as a developer I do not expect a method starting with `can-` to have a side-effect.